From 568ec0934db98033a5a11f4dd9fd23f7f8fa7b6b Mon Sep 17 00:00:00 2001
From: Zumeng Chen <zumeng.chen@windriver.com>
Date: Sun, 20 Jul 2014 15:13:58 +0800
Subject: [PATCH 139/263] arm: dts: add vpe/vip support hwmod

This patch comes from:

  git://git.omapzoom.org/kernel/omap.git

Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 arch/arm/boot/dts/dra7-evm.dts |  173 ++++++++++++++++++++++++++++++++++++++++
 1 files changed, 173 insertions(+), 0 deletions(-)

diff --git a/arch/arm/boot/dts/dra7-evm.dts b/arch/arm/boot/dts/dra7-evm.dts
index 0815781..cd1c28c 100644
--- a/arch/arm/boot/dts/dra7-evm.dts
+++ b/arch/arm/boot/dts/dra7-evm.dts
@@ -283,6 +283,110 @@
 		>;
 	};
 
+	vin1a_pins: pinmux_vin1a_pins {
+		pinctrl-single,pins = <
+			0x0DC	0x00040000 /* vin1a_clk0 OUTPUT | MODE0 */
+			0x0E4	0x00040000 /* vin1a_de0 OUTPUT | MODE0 */
+			0x0E8	0x00040000 /* vin1a_fld0 OUTPUT | MODE0 */
+			0x0EC	0x00040000 /* vin1a_hsync0	OUTPUT | MODE0 */
+			0x0F0	0x00040000 /* vin1a_vsync0 OUTPUT | MODE0 */
+			0x0F4	0x00040000 /* vin1a_d0	OUTPUT | MODE0 */
+			0x0F8	0x00040000 /* vin1a_d1	OUTPUT | MODE0 */
+			0x0FC	0x00040000 /* vin1a_d2	OUTPUT | MODE0 */
+			0x100	0x00040000 /* vin1a_d3 	OUTPUT | MODE0 */
+			0x104	0x00040000 /* vin1a_d4 	OUTPUT | MODE0 */
+			0x108	0x00040000 /* vin1a_d5	OUTPUT | MODE0 */
+			0x10C	0x00040000 /* vin1a_d6	OUTPUT | MODE0 */
+			0x110	0x00040000 /* vin1a_d7	OUTPUT | MODE0 */
+			0x114	0x00040000 /* vin1a_d8	OUTPUT | MODE0 */
+			0x118	0x00040000 /* vin1a_d9	OUTPUT | MODE0 */
+			0x11C	0x00040000 /* vin1a_d10	OUTPUT | MODE0 */
+			0x120	0x00040000 /* vin1a_d11	OUTPUT | MODE0 */
+			0x124	0x00040000 /* vin1a_d12	OUTPUT | MODE0 */
+			0x128	0x00040000 /* vin1a_d13	OUTPUT | MODE0 */
+			0x12C	0x00040000 /* vin1a_d14	OUTPUT | MODE0 */
+			0x130	0x00040000 /* vin1a_d15	OUTPUT | MODE0 */
+		>;
+	};
+
+	vin1a_d16_d23_pins: pinmux_vin1a_d16_d23_pins {
+		pinctrl-single,pins = <
+			0x134   0x00040000 /* vin1a_d16 OUTPUT | MODE0 */
+			0x138   0x00040000 /* vin1a_d17 OUTPUT | MODE0 */
+			0x13C   0x00040000 /* vin1a_d18 OUTPUT | MODE0 */
+			0x140   0x00040000 /* vin1a_d19 OUTPUT | MODE0 */
+			0x144   0x00040000 /* vin1a_d20 OUTPUT | MODE0 */
+			0x148   0x00040000 /* vin1a_d21 OUTPUT | MODE0 */
+			0x14C   0x00040000 /* vin1a_d22 OUTPUT | MODE0 */
+			0x150   0x00040000 /* vin1a_d23 OUTPUT | MODE0 */
+
+		>;
+	};
+
+	vin2a_pins: pinmux_vin2a_pins {
+		pinctrl-single,pins = <
+			0x154	0x00040000 /* vin2a_clk0 OUTPUT | MODE0 */
+			0x160	0x00040000 /* vin2a_hsync0 OUTPUT | MODE0 */
+			0x164	0x00040000 /* vin2a_vsync0 OUTPUT | MODE0 */
+			0x168	0x00040000 /* vin2a_d0	OUTPUT | MODE0 */
+			0x16c	0x00040000 /* vin2a_d1	OUTPUT | MODE0 */
+			0x170	0x00040000 /* vin2a_d2	OUTPUT | MODE0 */
+			0x174	0x00040000 /* vin2a_d3	OUTPUT | MODE0 */
+			0x178	0x00040000 /* vin2a_d4	OUTPUT | MODE0 */
+			0x17c	0x00040000 /* vin2a_d5	OUTPUT | MODE0 */
+			0x180	0x00040000 /* vin2a_d6	OUTPUT | MODE0 */
+			0x184	0x00040000 /* vin2a_d7	OUTPUT | MODE0 */
+		>;
+	};
+
+        vin3a_pins: pinmux_vin3a_pins {
+		pinctrl-single,pins = <
+			0x0E0	0x00040006 /* vin3a_clk0 OUTPUT | MODE0 */
+			0x1C0	0x00040005 /* vin3a_hsync0 OUTPUT | MODE0 */
+			0x1C4	0x00040005 /* vin3a_vsync0 OUTPUT | MODE0 */
+			0x134	0x00040006 /* vin3a_d0	OUTPUT | MODE0 */
+			0x138	0x00040006 /* vin3a_d1	OUTPUT | MODE0 */
+			0x13C	0x00040006 /* vin3a_d2	OUTPUT | MODE0 */
+			0x140	0x00040006 /* vin3a_d3 	OUTPUT | MODE0 */
+			0x144	0x00040006 /* vin3a_d4 	OUTPUT | MODE0 */
+			0x148	0x00040006 /* vin3a_d5	OUTPUT | MODE0 */
+			0x14C	0x00040006 /* vin3a_d6	OUTPUT | MODE0 */
+			0x150	0x00040006 /* vin3a_d7	OUTPUT | MODE0 */
+                >;
+        };
+
+	vin5a_pins: pinmux_vin5a_pins {
+		pinctrl-single,pins = <
+			0x374	0x00040009 /* vin5a_clk0 OUTPUT | MODE0 */
+			0x39c	0x00040009 /* vin5a_hsync0 OUTPUT | MODE0 */
+			0x3a0	0x00040009 /* vin5a_vsync0 OUTPUT | MODE0 */
+			0x398	0x00040009 /* vin5a_d0	OUTPUT | MODE0 */
+			0x394	0x00040009 /* vin5a_d1	OUTPUT | MODE0 */
+			0x390	0x00040009 /* vin5a_d2	OUTPUT | MODE0 */
+			0x38c	0x00040009 /* vin5a_d3 	OUTPUT | MODE0 */
+			0x388	0x00040009 /* vin5a_d4 	OUTPUT | MODE0 */
+			0x384	0x00040009 /* vin5a_d5	OUTPUT | MODE0 */
+			0x380	0x00040009 /* vin5a_d6	OUTPUT | MODE0 */
+			0x37c	0x00040009 /* vin5a_d7	OUTPUT | MODE0 */
+                >;
+        };
+
+	vin6a_pins: pinmux_vin6a_pins {
+		pinctrl-single,pins = <
+			0x298	0x00040007 /* vin6a_clk0 OUTPUT | MODE0 */
+			0x2B8	0x00040007 /* vin6a_hsync0 OUTPUT | MODE0 */
+			0x2B4	0x00040007 /* vin6a_vsync0 OUTPUT | MODE0 */
+			0x330	0x00040007 /* vin6a_d0	OUTPUT | MODE0 */
+			0x32C	0x00040007 /* vin6a_d1	OUTPUT | MODE0 */
+			0x328	0x00040007 /* vin6a_d2	OUTPUT | MODE0 */
+			0x324	0x00040007 /* vin6a_d3	OUTPUT | MODE0 */
+			0x310	0x00040007 /* vin6a_d4	OUTPUT | MODE0 */
+			0x30C	0x00040007 /* vin6a_d5	OUTPUT | MODE0 */
+			0x2F8	0x00040007 /* vin6a_d6	OUTPUT | MODE0 */
+			0x2F4	0x00040007 /* vin6a_d7	OUTPUT | MODE0 */
+		>;
+	};
+
 	cpsw_default: cpsw_default {
 		pinctrl-single,pins = <
 			/* Slave 1 */
@@ -832,6 +936,75 @@
 	voltage-tolerance = <1>;
 };
 
+&vip1 {
+	pinctrl-names = "default";
+	/*
+		Pinmux conflict between vin1a:16-23 & vin2a:0-8
+		Either use full 24bit of vin1a or use 16bit vin1a and 8bitvin2a
+		TO enable cam2, uncomment this
+		pinctrl-0 = <&vin1a_pins &vin2a>;
+	*/
+	pinctrl-0 = <&vin1a_pins &vin1a_d16_d23_pins>;
+
+	vin1a: port@0A {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		reg = <0>;
+	};
+
+	vin2a: port@1A {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		reg = <1>;
+	};
+};
+
+&vip2 {
+	pinctrl-names = "default";
+	/*
+		Pinmux conflict between vin3a and emac1
+		To enable vin3a, uncomment this
+		and remove Slave 2 section from cpsw_default_pins
+		pinctrl-0 = <&vin3a_pins>;
+	*/
+	pinctrl-0 = <&vin3a_pins>;
+
+	vin3a: port@0A {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		reg = <0>;
+	};
+
+	vin4a: port@1A {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		reg = <1>;
+	};
+};
+
+&vip3 {
+	pinctrl-names = "default";
+	/*
+		Pinmux conflict between vin6a and atl,radio,mcasp3 pins
+		To enable vin6a, uncomment this and
+		remove atl_pins, mcasp3_pins, radio_pins from dra7_pmx_core
+		pinctrl-0 = <&vin5a_pins &vin6a_pins>;
+	*/
+	pinctrl-0 = <&vin5a_pins>;
+
+	vin5a: port@0A {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		reg = <0>;
+	};
+
+	vin6a: port@1A {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		reg = <1>;
+	};
+};
+
 &cpu0 {
 	cpu0-supply = <&avs_mpu>;
 };
-- 
1.7.5.4

